affectedmanifests.go

```diff
@@ -0,0 +1,58 @@
+package claircore
+
+import (
+	"sort"
+	"sync"
+)
+
+// AffectedManifests describes a set of manifests affected by
+// a set of Vulnerabilities.
+type AffectedManifests struct {
+	mu sync.Mutex
+	// map of vulnerabilities keyed by the vulnerability's ID
+	Vulnerabilities map[string]Vulnerability `json:"vulnerabilities"`
+	// map associating a list of vulnerability ids keyed by the
+	// manifest hash they affect.
+	VulnerableManifests map[string][]string `json:"vulnerable_manifests"`
+}
+
+// NewAffectedManifests initializes a new AffectedManifests struct.
+func NewAffectedManifests() AffectedManifests {
+	return AffectedManifests{
+		Vulnerabilities:     make(map[string]Vulnerability),
+		VulnerableManifests: make(map[string][]string),
+	}
+}
+
+// Add will add the provided Vulnerability and Manifest digest
+// to the necessary maps.
+//
+// Add is safe to use by multiple goroutines.
+func (a *AffectedManifests) Add(v Vulnerability, digests ...Digest) {
+	a.mu.Lock()
+	a.Vulnerabilities[v.ID] = v
+	for _, d := range digests {
+		hash := d.String()
+		a.VulnerableManifests[hash] = append(a.VulnerableManifests[hash], v.ID)
+	}
+	a.mu.Unlock()
+}
+
+// Sort will sort each array in the VulnerableManifests map
+// by Vulnerability.NormalizedSeverity in Desc order.
+//
+// Sort should not be ran concurrently.
+func (a *AffectedManifests) Sort() {
```

Comment:
This should just take the lock, then there's not a way to use the struct unsafely.

Reply:
I didn't want to promote the use of Sort in multiple goroutines as it could potentially take some time and block others from doing work. what do you think about that?

Reply:
I think having some methods be concurrent-safe and some not is surprising. Either callers handle it or they don't, it shouldn't be a mix.